### PR TITLE
gulp + sass + generator docs + babel 6 + others optimizations

### PR DESCRIPTION
**Pull request by kevincaradant**

### CODE REVIEW

File: html-documentation/assets/anchor.js [hunk 1]
```diff
@@ -0,0 +1,197 @@
+/*!
```
Comment (maxime1992) [inline html-documentation/assets/anchor.js]: Do not push html-documentation please

File: karma.conf.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+module.exports = function(config) {
+  config.set({
+    browsers: ['PhantomJS'],
+    frameworks: ['jasmine'],
+    files: [
+   	  'node_modules/angular/angular.js',
```
Comment (maxime1992) [inline karma.conf.js]: Fix indent

File: src/app/app.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
+let app = angular.module('CUSTOM_NAME_OF_THE_APP', ['ngMockE2E', 'ngAnimate', 'ui.router', 'anim-in-out', 'ui.bootstrap', 'ngSanitize', 'pascalprecht.translate', 'angular-loading-bar']);
+let languages = ['en', 'fr'];
+
+(function () {
```
Comment (maxime1992) [inline src/app/app.js]: Use es6 block-scope

File: src/app/app.js [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+	* Enable / Disable debug for dev
+	* $compileProvider.debugInfoEnabled(true);
+	* $compileProvider.debugInfoEnabled(false);
+	*
+	* Enable/ Disable http caching
+	* $httpProvider.defaults.cache = true;
+	* $httpProvider.defaults.cache = true;
+	* @returns {undefinded} nothing
+	*/
+	app.config(function ($compileProvider, $httpProvider, $locationProvider, $stateProvider, $provide, $urlRouterProvider, $translateProvider) {
```
Comment (maxime1992) [inline src/app/app.js]: Use ES6 function

**Comment by maxime1992** [comment]
Thank you this is huge ! =)

File: src/app/app.js (hunk 3 above)
Comment (maxime1992) [inline src/app/app.js]: I gave a try with Babel 6 but it seems that it just let the braces. I opened a topic about that on Babel :  [http://discuss.babeljs.io/t/iife-block-only/211](http://discuss.babeljs.io/t/iife-block-only/211)

**Comment by kevincaradant** [comment]
You're welcome :),   if you or anybody see something great to add or refactor ,  i will be glad to hear that and continue to improve this template

**Comment by maxime1992** [comment]
Cool for the 2 commits, I think we're good to merge now !
As it's a huge improvement, I would like to keep it in a separate branch for some weeks before merging it into master, to make sure it's stable. I'm closing this PR now because it's pointing in master and I cannot change the branch (see issues about that probleme [here](http://blog.thoughtram.io/angular/2016/01/06/taking-advantage-of-observables-in-angular2.html)).
I'll create a "gulp" branch, please submit your PR again in this one. 
Thanks !

**Comment by kevincaradant** [comment]
Ok , i think it's good now on the gulp branch  :)

**Pull request closed without merging**